src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+    if (this->animIndex == EN_DT_ANIMATION_2 || this->animIndex == EN_DT_ANIMATION_5) {
+        morphFrames = 0.0f;
+    }
+
+    this->animEndFrame = (f32)Animation_GetLastFrame(sAnimations[this->animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->animEndFrame,
+                     sAnimationModes[this->animIndex], morphFrames);
+}
+
+void EnDt_UpdateVisualState(EnDt* this) {
+    s32 i = this->visualState;
+    s32* p = sVisualStateTable;
+    s32 v; // Permuter silliness
+
+    i = i * 4 + 1;
+    p += i;
+
+    i = *p;
+    v = *p;
+
+    // Extreme permuter silliness
+    goto dummy_label;
+dummy_label:;
+
+    EnDt_ChangeAnim(this, v);
+
+    this->eyeTexIndex = *(++p);
+    this->disableBlinking = *(++p);
+}
+
+void EnDt_UpdateCutsceneFocusTarget(EnDt* this) {
+    if (this->npcEnMuto != NULL && this->npcEnBaisen != NULL) {
+        EnMuto* npcEnMuto = (EnMuto*)this->npcEnMuto;
+        EnBaisen* npcEnBaisen = (EnBaisen*)this->npcEnBaisen;
+
+        switch (sCutsceneFocusTargetTable[this->textIdIndex]) {
+            case EN_DT_CS_FOCUS_TARGET_MUTO:
+                npcEnMuto->targetActor = this->npcEnMuto;
+                npcEnBaisen->targetActor = this->npcEnMuto;
+                this->targetActor = this->npcEnMuto;
+                break;
+            case EN_DT_CS_FOCUS_TARGET_BAISEN:
+                npcEnMuto->targetActor = this->npcEnBaisen;
+                npcEnBaisen->targetActor = this->npcEnBaisen;
+                this->targetActor = this->npcEnBaisen;
+                break;
+            case EN_DT_CS_FOCUS_TARGET_MAYOR:
+                npcEnMuto->targetActor = &this->actor;
+                npcEnBaisen->targetActor = &this->actor;
+                this->targetActor = &this->actor;
+                break;
+        }
+    }
+}
+
+void EnDt_SetupCutsceneNpcs(EnDt* this, PlayState* play) {
+    Actor* npc = play->actorCtx.actorLists[ACTORCAT_NPC].first;
+
+    while (npc != NULL) {
+        if (npc->id == ACTOR_EN_MUTO) {
+            this->npcEnMuto = npc;
+        } else if (npc->id == ACTOR_EN_BAISEN) {
+            this->npcEnBaisen = npc;
+        }
+        npc = npc->next;
+    }
+
+    EnDt_SetupRegularState(this, play);
+}
+
+void EnDt_SetupRegularState(EnDt* this, PlayState* play) {
+    EnDtActionFunc actionFn;
+    u32 textId;
+
+    this->textIdIndex = 0;
+
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_60_10)) {
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80)) {
+            this->textIdIndex = 21;
+            this->visualState = 5;
+            EnDt_UpdateVisualState(this);
+
+            this->actor.textId = sTextIds[this->textIdIndex];
+            Message_StartTextbox(play, this->actor.textId, &this->actor);
+            Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_WAIT);
+
+            actionFn = EnDt_UpdateMeetingCutscene;
+            this->npcState = EN_DT_NPC_STATE_IDLE;
+            this->actionFunc = actionFn;
```

Comment:
Interesting, feels like this could be eliminated somehow such that actionFn can be removed, since it happens twice in this function, almost want to leave a comment about it

Reply:
I thought this was odd as well, and it may very well not match the original. The branching is a bit complex in this one though, and I couldn't find another arrangement of the conditionals to match (without using labels anyway). If you have any suggestions, I'm all ears!

Reply:
```suggestion
            this->npcState = EN_DT_NPC_STATE_IDLE;
            this->actionFunc = EnDt_UpdateMeetingCutscene;
```
Seems to actually not be needed. You can just assign directly without the actionfunc temp.

Reply:
You're absolutely right. Fixed, and thanks for the suggestion.